GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -3309,6 +3344,61 @@ private void UpdateButtonStates()
                 : TranslatedStrings.ButtonPush;
         }
 
+        private void UICommands_PostRepositoryChanged(object sender, GitUIEventArgs e)
+        {
+            ThreadHelper.FileAndForget(async () =>
+                {
+                    await this.SwitchToMainThreadAsync();
+                    RescanChanges();
+                });
+        }
+
+        private bool ExecuteCommandStaged(int command)
```

Comment:
Please make it as local method of `StagedFileContext_Opening`

Reply:
Accepted